src/commands/command_factory.py

```diff
@@ -0,0 +1,28 @@
+from create_command import *
+
+# from set_command import *
+# from get_command import *
+# from delete_command import *
+
+
+class CommandFactory:
+    def __init__(self, input_adaptor):
+        self.input_adaptor = input_adaptor
+        self.validate()
+        
+    def create(self):
+        cmd = str(self.input_adaptor.command).lower()
+        if cmd == "create":
+            return CreateCommand(self.input_adaptor.schema)
+#        elif cmd == "set":
+#            return SetCommand(self.input_adaptor.database, self.input_adaptor.table, self.input_adaptor.primary_key, self.input_adaptor.parameter, self.input_adaptor.value)
+#        elif cmd == "get":
+#            return GetCommand(self.input_adaptor.database, self.input_adaptor.table, self.input_adaptor.primary_key)
+#        elif cmd == "delete":
+#            return DeleteCommand(self.input_adaptor.database, self.input_adaptor.table, self.input_adaptor.primary_key)
+
+    def validate(self):
+        if self.input_adaptor.command == "none" or self.input_adaptor.command is None:
+            raise NoParameterError("No command was entered")
```

Comment:
I don’t think you need this here.
You need this when you are creating command object.

In line #17 you need to have else raise wrongParamwterError

Reply:
For line #17, the validate function already checks if the input is one of the available inputs so no need to check there again